pkg/scheduler/algorithm/predicates/metadata.go

```diff
@@ -185,6 +185,9 @@ func (topologyPairsMaps *topologyPairsMaps) removePod(deletedPod *v1.Pod) {
 }
 
 func (topologyPairsMaps *topologyPairsMaps) appendMaps(toAppend *topologyPairsMaps) {
+	if toAppend == nil || len(toAppend.podToTopologyPairs) == 0 {
```

Comment:
The length check is not needed since ranging over an empty/nil map skips the loop.

Reply:
Agree. Done.